### DISCUSSION TITLE
Promise JSON & send another request

### DISCUSSION THREAD

**Discussion by kravcik** [jiripudil/Naja#337] [Support] [answered]
At first thanks for nice piece of code!

Im probably dumb, I have this code and works ok. But when Im trying send AJAX instead of `console.log("X")` - like second code. Handle in presenter (nette) start redirect - but nothing happens (console write error json <).

What Im missing? Im trying migrate from `nette.ajax.js`.

```
naja.makeRequest('GET', 'http://webproject.local/front/player/json')
	.then(payload =>
	{
		$("#autocomplete").autocomplete({
			source: payload,
			treshold: 2,
			maximumItems: 10,
			onSelectItem: function(item)
			{
				console.log("X");						
			}
		});
	});
```

```
...
onSelectItem: function(item)
{
	naja.makeRequest('GET', {link search!, playerId => 'PLAYER_ID'}.replace('PLAYER_ID', item.value));					
}
```

Thanks for tips

**Comment by jiripudil** [comment, marked as answer]
Hi, "error json <" suggests that the response body contains HTML code instead of JSON payload which Naja expects. I'd start investigating from there.

**Reply by kravcik** [reply]
Thanks for answer. I guess I missing some knowledge about naja behavior. Maybe problem is in presenter handler:

```
pf handleSearh($playerId)
{
  $this->redirect(':Front:Player:somethign');
}
```

I tried also send `->sendPayload(forceRedirect, url etc.)` instead of `->redirect()`. But no redirect happens (I saw hi in Tracy, but page stay same).

So maybe few questions:
1. Am I too dumb to use Naja?
2. Its possible arrange some lesson with you online ($$$)?
3. Can I somehow help on this issue (https://github.com/jiripudil/Naja/issues/211) - I guess a lot of people has working AJAX with oldschool nette.ajax.js and its really hard change it for Naja - even if they want use Naja.

**Reply by kravcik** [reply]
So I get it now where is problem, I didnt think about this init()...

works:
`naja.makeRequest(method, url, data, {forceRedirect: true})`

not works:
`naja.initialize({forceRedirect: true})`

Is there way to set default redirects?

**Reply by jiripudil** [reply]
Of course you're not dumb. You're getting to know a new library, it is only natural that you don't understand everything perfectly from the beginning.

As you seem to have learned, Naja follows redirects unless you tell it not to via the `forceRedirect` option. But I'm afraid I don't know why setting it in `initialize` doesn't work for you 😟 Could you please try and reproduce the error in the [sandbox](https://github.com/jiripudil/Naja-sandbox)? It works for me there, so I'm thinking there might be something elsewhere in your code interfering with Naja.

**Reply by kravcik** [reply]
Its help replace "old standard fetch" from chain of ajax. Thanks for advices!

Still if possible to get some lessons (for money) about Naja (and new JS approach)... if you planning something contact me on email (in github profile).
